### PR TITLE
Fix conditional dependency resolution when runtime-deployment dependencies are not mirrored

### PR DESCRIPTION
**Pull request by aloubyansky**
Fixes an issue reported in https://quarkusio.zulipchat.com/#narrow/channel/187038-dev/topic/conditional.20dependency.20with.20multiple.20layer.20of.20dependency/near/482870695

Tests included.

Also includes doc adjustments in the second commit pointed out in https://github.com/quarkusio/quarkus/pull/44424

### CODE REVIEW

File: independent-projects/bootstrap/core/src/test/java/io/quarkus/bootstrap/resolver/TsArtifact.java [hunk 1]
```diff
@@ -179,6 +182,33 @@ public TsArtifact addDependency(TsDependency dep) {
...
+     * @param dep dependency to add
+     * @return this artifact
+     */
+    public TsArtifact addFirstDependency(TsDependency dep) {
+        if (deps.isEmpty()) {
+            deps = new ArrayList<>();
+            deps.add(dep);
+        } else {
+            deps.add(dep);
+            Collections.rotate(deps, 1);
```
Comment (geoand) [inline independent-projects/bootstrap/core/src/test/java/io/quarkus/bootstrap/resolver/TsArtifact.java:197]: TIL :)

File: independent-projects/bootstrap/core/src/test/java/io/quarkus/bootstrap/resolver/test/ConditionalDependenciesDevModelTestCase.java [hunk 2]
```diff
@@ -13,7 +13,7 @@ public class ConditionalDependenciesDevModelTestCase extends CollectDependencies
     @Override
     protected BootstrapAppModelResolver newAppModelResolver(LocalProject currentProject) throws Exception {
         var resolver = super.newAppModelResolver(currentProject);
-        resolver.setIncubatingModelResolver(false);
+        //resolver.setIncubatingModelResolver(false);
```
Comment (geoand) [inline independent-projects/bootstrap/core/src/test/java/io/quarkus/bootstrap/resolver/test/ConditionalDependenciesDevModelTestCase.java:16]: Is this needed?

File: core/deployment/src/test/java/io/quarkus/deployment/conditionaldeps/DependencyConditionMatchesConditionalDependencyTest.java [hunk 3]
```diff
@@ -5,14 +5,23 @@
...
+import io.quarkus.bootstrap.resolver.maven.workspace.LocalProject;
 import io.quarkus.deployment.runnerjar.BootstrapFromOriginalJarTestBase;
 import io.quarkus.maven.dependency.ResolvedDependency;
 
 public class DependencyConditionMatchesConditionalDependencyTest extends BootstrapFromOriginalJarTestBase {
 
+    @Override
+    protected BootstrapAppModelResolver newAppModelResolver(LocalProject currentProject) throws Exception {
+        var resolver = super.newAppModelResolver(currentProject);
+        //        resolver.setIncubatingModelResolver(true);
```
[the remark this answers is not in the document]
Comment (aloubyansky) [reply core/deployment/src/test/java/io/quarkus/deployment/conditionaldeps/DependencyConditionMatchesConditionalDependencyTest.java:21]: There is a global switch to enable the incubating impl but it's more convenient to do it this way when testing in an IDE. I can remove it in a couple of subsequent PRs (on the way).

[the remark this answers is not in the document]
Comment (geoand) [reply core/deployment/src/test/java/io/quarkus/deployment/conditionaldeps/DependencyConditionMatchesConditionalDependencyTest.java:21]: Okay, thanks

**Review by geoand — APPROVED** [review]

**Pull request merged**